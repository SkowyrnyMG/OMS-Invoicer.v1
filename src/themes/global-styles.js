import { createGlobalStyle } from 'styled-components';

const GlobalStyles = createGlobalStyle`
*,
*::before,
*::after {
  box-sizing: inherit;
  margin: 0;
  padding: 0;
}

html {
  font-size: 62.5%;
}

body {
  font-size: 1.6rem;
}

`;

export default GlobalStyles;
