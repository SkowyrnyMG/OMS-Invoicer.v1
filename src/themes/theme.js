export const theme = {
  color: {
    bg: '#F8F8FF',
    bgSecondary: '#FFFFFF',
    mainFont: '#000',
    transparentMain: '#446DF644',
    secondaryFont: '#8F8F8F',
    tertiaryFont: '#FFFFFF',
    devider: '#F0F0FD',
    primary: '#446DF6',
    success: '#00A896',
    danger: '#F25C54',
    error: '#DA2442',
  },
  fontSize: {
    heading: '64px',
    headingSmall: '34px',
    cta: '24px',
    l: '20px',
    regular: '18px',
    ms: '16px',
    s: '14px',
  },
  fontWeight: {
    light: 300,
    regular: 400,
    bold: 700,
  },
  shadow: {
    around: '0 0 10px -4px rgba(0, 0, 0, 0.25)',
    bottom: '0 4px 5px rgba(0, 0, 0, 0.1)',
    right: '4px 0 5px rgba(0, 0, 0, 0.1)',
  },
  defaultSizes: {
    appBody: '1268px',
  },
};
