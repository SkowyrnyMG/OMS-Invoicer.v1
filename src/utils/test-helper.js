import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
