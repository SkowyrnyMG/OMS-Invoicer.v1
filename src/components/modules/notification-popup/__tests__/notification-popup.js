import React from 'react';

import { renderWithReduxRouter, screen } from 'utils/tests/test-helper';
import { theme } from 'themes/theme';
import NotificationPopup from '../notification-popup';

const successMsg =
  'Success! The popup informs you that something was done correcty!';

const errorMsg = 'Error! The popup informs you that something went wrong!';

describe('NotificaionPopup', () => {
  test('should display succes notification in the doucment with correct styles', () => {
    renderWithReduxRouter(
      <div>
        <NotificationPopup successRegexp='Success!'>
          {successMsg}
        </NotificationPopup>
        <button>submit</button>
      </div>,
    );

    const NotificationPopupNode = screen.getByTestId('notification-popup');
    const NotificationSpanNode = screen.getByText(successMsg);

    expect(NotificationPopupNode).toBeInTheDocument();
    expect(NotificationSpanNode).toHaveStyle(`color: ${theme.color.success}`);
    expect(NotificationPopupNode).toMatchSnapshot();
  });

  test('should display error notification in the doucment with correct styles', () => {
    renderWithReduxRouter(
      <div>
        <NotificationPopup successRegexp='Success!'>
          {errorMsg}
        </NotificationPopup>
        <button>submit</button>
      </div>,
    );

    const NotificationPopupNode = screen.getByTestId('notification-popup');
    const NotificationSpanNode = screen.getByText(errorMsg);

    expect(NotificationPopupNode).toBeInTheDocument();
    expect(NotificationSpanNode).toHaveStyle(`color: ${theme.color.danger}`);
    expect(NotificationPopupNode).toMatchSnapshot();
  });
});
