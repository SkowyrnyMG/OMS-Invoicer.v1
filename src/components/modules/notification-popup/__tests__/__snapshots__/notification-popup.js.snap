// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificaionPopup should display error notification in the doucment with correct styles 1`] = `
.c1 {
  margin-bottom: 1rem;
  width: 100%;
  font-size: 14px;
  color: #F25C54;
}

.c0 {
  margin: -2rem 0 2rem;
}

<div
  class="c0"
  data-testid="notification-popup"
>
  <span
    class="c1"
    style="opacity: 0; visibility: hidden;"
  >
    Error! The popup informs you that something went wrong!
  </span>
</div>
`;

exports[`NotificaionPopup should display succes notification in the doucment with correct styles 1`] = `
.c1 {
  margin-bottom: 1rem;
  width: 100%;
  font-size: 14px;
  color: #00A896;
}

.c0 {
  margin: -2rem 0 2rem;
}

<div
  class="c0"
  data-testid="notification-popup"
>
  <span
    class="c1"
    style="opacity: 0; visibility: hidden;"
  >
    Success! The popup informs you that something was done correcty!
  </span>
</div>
`;
